pkg/scheduler/DESIGN.md

```diff
@@ -8,24 +8,32 @@ the queuing logic is isolated into a "tree queue" structure and its associated q
 
 ## Tree Queue: What and Why
 
-The "tree queue" structure serves the purpose of a discrete priority queue.
-The requests are split into many queues, each of which is located at a leaf node in the tree structure.
+The "tree queue" structure serves as a discrete priority queue.
+Requests are split into many queues, each of which is located at a leaf node in the tree structure.
 
 The tree structure enables some of the specific requirements of our queue selection algorithms:
 
-- we must select a queue to dequeue from based on two independent algorithms, each with their own state
-- there is a hierarchy of importance between the two algorithms - one is primary, the other secondary
-- one of the algorithms (tenant-querier shuffle shard) can reject all queue options presented to it,
-  requiring us to return back up to the previous level of queue selection to continue searching.
+1. The system must select a queue to dequeue from based on two independent algorithms, each with their own state:
```

Comment:
I stopped trying to be so generic about what algorithms we are talking about in the requirements here, and just listed the actual differences by algorithm